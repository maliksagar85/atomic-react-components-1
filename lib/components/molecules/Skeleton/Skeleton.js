/**
 *
 * Skeleton
 *
 */
import React from 'react';
import styled from 'styled-components';
import classnames from 'classnames';
import PropTypes from 'prop-types';
import styles from './Skeleton.style';

const Circle = ({ className, width, marginBottom }) => (
  <div className={classnames('circle', className)} style={{ width, marginBottom, height: width }} />
);

const Bar = ({ className, height, width, marginBottom }) => (
  <div className={classnames('bar', className)} style={{ height, width, marginBottom }} />
);

const Skeleton = ({ children, className }) => (
  <div className={classnames('skeleton', className)}>{children}</div>
);

Skeleton.Circle = Circle;
Skeleton.Bar = Bar;

Skeleton.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
};

Circle.propTypes = {
  className: PropTypes.string,
  marginBottom: PropTypes.string,
  width: PropTypes.string,
};

Bar.propTypes = {
  className: PropTypes.string,
  marginBottom: PropTypes.string,
  width: PropTypes.string,
  height: PropTypes.string,
};

Skeleton.defaultProps = {
  className: '',
};

Bar.defaultProps = {
  className: '',
  height: '18px',
  width: '100%',
  marginBottom: '0',
};

Circle.defaultProps = {
  className: '',
  width: '100px',
  marginBottom: '20px',
};

const StyledSkeleton = styled(Skeleton)`
  ${styles};
`;

export default StyledSkeleton;

export { Skeleton as SkeletonVanilla };
