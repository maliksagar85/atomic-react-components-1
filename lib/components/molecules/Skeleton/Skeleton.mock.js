const defaultConfig = {
  className: 'test-line',
  height: '20px',
  width: '100%',
  marginBottom: '0',
};

const circleConfig = {
  className: 'test-circle',
  width: '100px',
  marginBottom: '25px',
};

const accordionConfig = {
  className: 'accordion',
  height: '50px',
  width: '100%',
  marginBottom: '2px',
};

const tabsConfig = {
  className: 'tabs',
  height: '46px',
  width: '84px',
  marginBottom: '1px',
};

const tabsContentConfig = {
  className: 'tabs',
  height: '84px',
  width: '936px',
  marginBottom: '10px',
};

export { defaultConfig, circleConfig, accordionConfig, tabsConfig, tabsContentConfig };
