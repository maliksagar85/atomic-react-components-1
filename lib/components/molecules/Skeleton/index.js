export { default } from './Skeleton';
export { SkeletonVanilla } from './Skeleton';
