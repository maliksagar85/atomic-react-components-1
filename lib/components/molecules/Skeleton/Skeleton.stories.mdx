import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';

import {
  defaultConfig,
  circleConfig,
  accordionConfig,
  tabsConfig,
  tabsContentConfig,
} from './Skeleton.mock';

import Skeleton, {SkeletonVanilla} from '.';


<Meta title="Molecules/Skeleton" component={SkeletonVanilla} />

### Default Skeleton
<Preview>
  <Story name="Default Skeleton">
  <Skeleton {...defaultConfig}>
      <Skeleton.Bar {...defaultConfig} />
      <Skeleton.Bar {...defaultConfig} />
      <Skeleton.Circle {...circleConfig} />
      <Skeleton.Circle {...circleConfig} />
      <Skeleton.Bar {...defaultConfig} />
    </Skeleton>
  </Story>
</Preview>

### Accordion Skeleton
<Preview>
  <Story name="Accordion Skeleton">
      <Skeleton>
      <Skeleton.Bar {...defaultConfig} />
      <Skeleton.Bar {...accordionConfig} />
      <Skeleton.Bar {...defaultConfig} />
      <Skeleton.Bar {...accordionConfig} />
    </Skeleton>
  </Story>
</Preview>

### Tabs Skeleton
<Preview>
  <Story name="Tabs Skeleton">
<Skeleton>
      <div style={{ display: 'block', background: 'none', animation: 'none' }}>
        <Skeleton.Bar {...tabsConfig} />
        <Skeleton.Bar {...tabsConfig} />
      </div>
      <Skeleton.Bar {...tabsContentConfig} />
    </Skeleton>
  </Story>
</Preview>



