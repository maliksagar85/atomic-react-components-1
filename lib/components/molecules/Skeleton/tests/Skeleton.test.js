import React from 'react';
import { mount } from 'enzyme';

import Skeleton from '../index';

describe('<Skeleton />', () => {
  let SkeletonComponent = '';
  beforeEach(() => {
    SkeletonComponent = mount(
      <Skeleton>
        <Skeleton.Bar />
        <Skeleton.Circle />
      </Skeleton>
    );
  });

  test('should render the <Skeleton> /', () => {
    expect(SkeletonComponent.exists()).toBe(true);
  });

  test('should render <div/>', () => {
    expect(SkeletonComponent.find('div')).toBeTruthy();
  });

  test('should render a skeleton line', () => {
    expect(SkeletonComponent.find('.bar')).toBeTruthy();
  });

  test('should render a skeleton circle', () => {
    expect(SkeletonComponent.find('.circle')).toBeTruthy();
  });

  test('should render correctly', () => {
    expect(SkeletonComponent).toMatchSnapshot();
  });
});
