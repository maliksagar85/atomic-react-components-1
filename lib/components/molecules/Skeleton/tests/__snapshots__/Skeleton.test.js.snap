// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Skeleton /> should render correctly 1`] = `
<Skeleton__StyledSkeleton>
  <Skeleton
    className="Skeleton__StyledSkeleton-sc-1oak7zh-0 iDRFJU"
  >
    <div
      className="skeleton Skeleton__StyledSkeleton-sc-1oak7zh-0 iDRFJU"
    >
      <Bar
        className=""
        height="18px"
        marginBottom="0"
        width="100%"
      >
        <div
          className="bar"
          style={
            Object {
              "height": "18px",
              "marginBottom": "0",
              "width": "100%",
            }
          }
        />
      </Bar>
      <Circle
        className=""
        marginBottom="20px"
        width="100px"
      >
        <div
          className="circle"
          style={
            Object {
              "height": "100px",
              "marginBottom": "20px",
              "width": "100px",
            }
          }
        />
      </Circle>
    </div>
  </Skeleton>
</Skeleton__StyledSkeleton>
`;
